pkg/storage/stores/local/filescollection.go

```diff
@@ -0,0 +1,132 @@
+package local
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"go.etcd.io/bbolt"
+)
+
+type downloadedFile struct {
+	mtime  time.Time
+	boltdb *bbolt.DB
+}
+
+// FilesCollection holds info about shipped boltdb index files by other uploaders(ingesters).
+// It is used to hold boltdb files created by all the ingesters for same period i.e with same name.
+// In the object store files are uploaded as <boltdb-filename>/<uploader-id> to manage files with same name from different ingesters.
+// Note: FilesCollection does not manage the locks on mutex to allow users of it to
+// do batch operations or single operation or operations requiring intermittent locking.
+// Note2: It has an err variable which is set when FilesCollection is in invalid state due to some issue.
+// All operations which try to access/update the files except cleanup returns an error if set.
+type FilesCollection struct {
+	sync.RWMutex
```

Comment:
I think I would be happier to see this as a named private field, as written because FilesCollection is public this would allow anyone to manipulate the mutex.

It's also harder to track the uses of the mutex when it doesn't have a name